### DISCUSSION TITLE
What sorting criteria are used to search for the most frequently viewed data?

### DISCUSSION THREAD

**Discussion by anesvel** [geonetwork/geonetwork-ui#691] [General]
We'd like to know more about the popularity and relevance filters. What criteria are applied? 
In practical terms, we need to apply a filter based on the number of times a dataset is consulted. 
Which filter should we apply? 
Where would you store the data? raiting? score? ...

**Comment by fgravin** [comment]
Hi @anesvel, welcome in the community.

So let me provide some inputs about the sort
- The **releavancy** sorting option in Elasticsearch (ES) uses the internal Elasticsearch `score` property to sort the results. When you construct a query in Elasticsearch, it calculates a score for each document based on how well it matches the query criteria. The documents are then sorted by their scores in descending order, meaning that documents with higher scores are ranked higher in the search results.

- The **popularity** sorting option in the datahub is based on the number of times a record has been saved as a favorite. The more times a record is selected as a favorite, the higher its popularity ranking will be in the datahub.

None of those sort options answer to your need. Nevertheless, all records have a field `popularity` in the index, which is incremented every time a record is opened. This means that we could have a sort option based on that property. This would require some development though.

Todo:
- Add an option in `SortByEnum`
- Use this option in the `SortByComponent`



> Where would you store the data? rating? score? ...

I don't understand your question.

**Comment by anesvel** [comment]
Hi @fgravin 

Thanks for the clarifications.
For the moment, the proposed solution suits us.

Thank you for your reactivity!
